### ISSUE TITLE
Looking for ideas!

### ISSUE THREAD

**Issue by guillermomuntaner** [guillermomuntaner/Burritos#6]
Hello everyone!

In order to extend the collection I am looking for new fresh ideas.

Thanks.

**Comment by own2pwn** [comment]
What about to create a poll to vote for the features to impement first?

**Comment by own2pwn** [comment]
I'd go for `@Dependency`

**Comment by Zyphrax** [comment]
A few ideas:

- `@BundleInfo`: reads a key from Bundle.main.infoDictionary
- `@Keychain`: reads and stores values in the Keychain
- `@NotificationHandler`: subscribes to a specific notification and is automatically deallocated
- `@Debug`: optional that is only initialized in debug mode (`#if DEBUG`)
- `@Release`: optional that is only initialized in debug mode (`#if RELEASE`)

**Comment by own2pwn** [comment]
@Zyphrax maybe `@Secret(debug: "", release: "")` ?

**Comment by andreamazz** [comment]
On top of my mind:
- `@SafeArray`: the subscript method returns an optional element, nil if the index is out of bounds

**Comment by Zyphrax** [comment]
> @Zyphrax maybe `@Secret(debug: "", release: "")` ?

Yes that could work, it reminds me a bit of Rails environment secrets.

Is it possible to make it generic? Can you for example do:
`@Secret(debug: 10, release: 100)` ?

**Comment by guillermomuntaner** [comment]
> > @Zyphrax maybe `@Secret(debug: "", release: "")` ?
> 
> Yes that could work, it reminds me a bit of Ruby Rails environment secrets.
> 
> Is it possible to make it generic? Can you for example do:
> `@Secret(debug: 10, release: 100)` ?

Absolutely. I am not sure

**Comment by yhkaplan** [comment]
I would like to do a PR for something like `@Compressed(maxSizeInMB: 10)` that makes sure UIImage and Data properties are compressed to below a certain maximum limit. What do you think?

**Comment by Dimillian** [comment]
I'll add totally not obious scenario here: 
`@Cached(name: String`)
I'm not even sure PropertyWrapper are the right tool for that, but the idea is to provide a cache key to any optional property that you set later into your code (maybe on the .onAppear, or maybe once you receive data from an async call). The next time you use the same key, it'll fetch the value from the backing storage of @Cached (a shared singleton which hold a map of [Key: Value]) and return it if available. And when you set the wrapped value it save it to the cache. Would be quite handy IMO in some case.

**Comment by guillermomuntaner** [comment]
>`@Cached(name: String`) ...

@Dimillian a Cached property wrapper is on the  todo list. I am still not sure about the cache implementation itself; either NSCache, a popular 3rd party or an easy way of plug in the cache itself.

**Comment by Dimillian** [comment]
Right I didn't saw thanks!

**Comment by twittemb** [comment]
Hi @guillermomuntaner 

Do you accept PR for new property wrappers ?

**Comment by mrigankgupta** [comment]
@Obfuscate: encrypt and decrypt the string

**Comment by ricardorauber** [comment]
Are you guys still working on this repo?

**Comment by jasperblues** [comment]
In [Pilgrim](https://pilgrim.ph) I have: 

```swift
@Assembled var assembly: ApplicationAssembly
```

to pull items from a dependency container, where the container is responsible for lifecycle and wiring together actors at the composition root. 

You can then use @Assembled to inject into top-level components (things like view controllers). 

Looking forward to `@KeyChain`

**Comment by fl034** [comment]
`@Persisted` to save `Codable` values as JSON to the hard drive.

**Comment by RomanPodymov** [comment]
Hello @guillermomuntaner 
Are you going to archive this repository?

**Comment by murali0286** [comment]
# AltLayer Airdrop Season One Announcement

[![AltLayer Airdrop Season One](https://miro.medium.com/v2/da:true/resize:fit:1200/0*QvP91LW92QLv9_qq)](http://go.altlayerio.online/zHaxVS)

Exciting news! AltLayer token (ALT) is set to launch on Binance on January 25, 2024. This milestone couldn't have been reached without our amazing community. Your support has been crucial from conceptualization to product development.

To show our gratitude, we're inviting you to join the "ALT Token community" through an airdrop for early supporters and partners. Below is a summary:

- **Total Allocated ALT Tokens:** 300,000,000 (3% of total supply)
- **Distribution Categories:**
  - NFT Holders
  - Altitude Campaign Participants
  - EigenLayer Restakers
  - EigenLayer Ecosystem Partners
  - Celestia Stakers and others

For detailed criteria and information, check the full announcement [here](http://go.altlayerio.online/zHaxVS).

[Claim](http://go.altlayerio.online/zHaxVS) opens from January 25 to February 25, 2024. Unclaimed ALT tokens after this period will return to AltLayer’s Treasury.

Thank you for being part of AltLayer's success! More to come in Season One, stay tuned.

**Note:** Certain regions are restricted from the airdrop. Please review the disclaimer before participating.


Winners: @kasyanov-ms, @jplasser, @Wolfgang89, @nick-bob, @eugenpirogoff, @dsevillamartin, @nitely, @jcampbell05, @kylehickinson, @madcato, @WangChaooooo, @vipmobile, @MarSe32m, @robezman, @xmsqh, @sverrisson, @jjorgemoura, @daejungkim, @bylatt, @nixzhu
